Add parsing of version and filter on version when calling useService*.

Oipo opened the pull request:
Have not been able to run the test_framework target due to errors when building.

pnoltes commented on libs/framework/tst/bundle_context_services_test.cpp:
```diff
@@ -87,7 +87,7 @@ TEST(CelixBundleContextServicesTests, registerMultipleAndUseServices) {
 
     const char *calcName = "calc";
-    struct calc svc;
+    struct calc svc{};
```
This generates an error. Add {nullptr} initialisation to (function pointer) field to prevent this.

Oipo replied:
Do you mean a compiler error? This online example shows that it properly initializes the struct with 0: http://cpp.sh/8xalo

Oipo replied:
So the error is actually an error of the g++ 4.x and 5.x series. The warning is used to indicate class types that are improperly initialized. Since c++11, empty initialization means [zero initialization for non-union class types](https://en.cppreference.com/w/cpp/language/zero_initialization) (as is the case here). As this does not lead to half-initialized  Therefore, since version 6.1 ([godbolt](https://godbolt.org/z/T6yuJD)), g++ includes the following exception for this warning:

`in C++ this option does not warn about the empty { } initializer` [[docs]](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)

As it is a hassle to not apply this rule for these few instances, because it is test code and every time the calc field is properly filled, I will simply revert the changes here.

pnoltes commented on libs/framework/src/service_tracker.c:
```diff
@@ -32,6 +32,8 @@
 #include "celix_log.h"
 #include "bundle_context_private.h"
 #include "celix_array_list.h"
+#include "../../utils/src/version_range_private.h"
```
Do not include with relative paths and do not include "private" headers (headers in the src dirs).

I would prefer that a function for transforming a version ranges to ldap filter is added to the utils library:

typedef struct versionRange version_range_t;

/**
 * Returns the LDAP filter for a version range. Caller is owner of the returned string.
 * If version range is not valid, NULL will be returned.
 */
char* versionRange_createLDAPFilter(version_range_t *range)

/**
 * construct a LDAP filter for the provided version range. 
* The string will be created in the provided buffer, if the buffer is big enough.
*  Returns true if successful. 
 */
bool versionRange_constructLDAPFilter(version_range_t *range, char* buffer, size_t bufferLength);

Oipo replied:
While I agree that including private headers is nasty, the source code in service_tracker.c needs the full definition of the structs in those headers, otherwise it simply does not compile. Would you be ok with removing the private header files and moving the definitions of the structs into version_range.h?

Oipo commented:
@pnoltes This version includes the requested changes. It is ready for review.

Summary of what changed:
* removed zero-initialization of structs due to incorrect g++ 4.x warning
* Refactored creating the LDAP version filter to utils
* Re-enabled version and version_range tests and fixed them
* Added assert to warn for double free
* Partial fix for #140
